### ISSUE TITLE
Calling repository method  returns error since 1.24

### ISSUE THREAD

**Issue by 24HOURSMEDIA** [zenstruck/foundry#409]
Since version 1.24, I get this error in some test cases.

```
   ┐
   ├ TypeError: Zenstruck\Foundry\RepositoryProxy::Zenstruck\Foundry\{closure}(): Argument #1 ($o) must be of type object, array given
   │
   ╵ /var/www/html/vendor/zenstruck/foundry/src/RepositoryProxy.php:377
   ╵ /var/www/html/vendor/zenstruck/foundry/src/RepositoryProxy.php:30
```

The issue seems to be in RepositoryProxy::proxyResult(mixed $result) where a close expects an object where an array is provided.

Test code:

```
SomeFactory::repository()->collectIds()
```

Repository code:

```
class SomeRepository {
  public function collectIds(
        DateTimeInterface $publishedFrom,
        DateTimeInterface $publishedUntil
    ): array {
         return $qb->getQuery()->getScalarResult();
  }
}
```

**Comment by nikophil** [comment]
hello @24HOURSMEDIA 

you're right, I've fixed it here: https://github.com/zenstruck/foundry/pull/410

**Comment by nikophil** [comment]
hey @24HOURSMEDIA I've released the fix in [1.28.0](https://github.com/zenstruck/foundry/releases/tag/v1.28.0)
